tests/test_anker.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2021 Chorus One AG
+# SPDX-License-Identifier: GPL-3.0
+
+"""
+This script calls 'solana' and 'solido' to confirm that functionality works.
+
+It exits with exit code 0 if everything works as expected, or with a nonzero
+exit code if anything fails. It expects a test validator to be running at at the
+default localhost port, and it expects a keypair at ~/.config/solana/id.json
+that corresponds to a sufficiently funded account.
+"""
+import os
+import json
+
+from util import (
+    TestAccount,
+    create_spl_token_account,
+    create_test_account,
+    create_vote_account,
+    get_solido_program_path,
+    multisig,
+    solana,
+    solana_program_deploy,
+    solido,
+    spl_token,
+)
+
+print('Creating test accounts ...')
+os.makedirs('tests/.keys', exist_ok=True)
+test_addrs = [
+    create_test_account('tests/.keys/test-key-1.json'),
+    create_test_account('tests/.keys/test-key-2.json'),
+]
+print(f'> {test_addrs}')
+
+treasury_account_owner = create_test_account('tests/.keys/treasury-key.json')
+print(f'> Treasury account owner:      {treasury_account_owner}')
+
+developer_account_owner = create_test_account('tests/.keys/developer-fee-key.json')
+print(f'> Developer fee account owner: {developer_account_owner}')
+
+
+print('\nSetting up UST mint ...')
+ust_mint_address = create_test_account('tests/.keys/ust_mint_address.json', fund=False)
+spl_token('create-token', 'tests/.keys/ust_mint_address.json')
+print(f'> UST mint is {ust_mint_address.pubkey}.')
+
+print('\nUploading Multisig program ...')
+multisig_program_id = solana_program_deploy(
+    get_solido_program_path() + '/serum_multisig.so'
+)
+print(f'> Multisig program id is {multisig_program_id}.')
+
+print('\nUploading Solido program ...')
+solido_program_id = solana_program_deploy(get_solido_program_path() + '/lido.so')
+print(f'> Solido program id is {solido_program_id}.')
+
+print('\nUploading Anker program ...')
+anker_program_id = solana_program_deploy(get_solido_program_path() + '/anker.so')
+print(f'> Anker program id is {anker_program_id}.')
+
+print('\nCreating new multisig ...')
+multisig_data = multisig(
+    'create-multisig',
+    '--multisig-program-id',
+    multisig_program_id,
+    '--threshold',
+    '1',
+    '--owners',
+    ','.join(t.pubkey for t in test_addrs),
+)
+multisig_instance = multisig_data['multisig_address']
+multisig_pda = multisig_data['multisig_program_derived_address']
+print(f'> Created instance at {multisig_instance}.')
+
+
+def approve_and_execute(
```

Comment:
Can we use `approve_and_execute` from `prepare_test_environment`?
This function seems to only execute the tx, not doing the approve part, but it seems not to be called anywhere.

Reply:
It does not need to be approved in this case, because I set up the multisig with a threshold of 1 for these tests, and the proposer automatically approves. But yeah I think we can reuse `prepare_test_environment` instead :+1:

Reply:
Ah wait it’s a bit more subtle than this, but we can make it a higher-order function.